File: FrontEnd/lib/domain/interfaces/usecases/plants/i_search_species_use_case.dart
/// @file i_search_species_use_case.dart
/// @description Interfaz del caso de uso para buscar especies de plantas.
/// @module Plants
/// @layer Domain
library;

import '../../../entities/plant_species.dart';

abstract interface class ISearchSpeciesUseCase {
  /// Busca especies por [query]. Devuelve lista vacía si no hay resultados.
  Future<List<PlantSpecies>> execute(String query, {int limit = 20});
}

File: BackEnd/src/domain/interfaces/usecases/plants/ISearchSpeciesUseCase.ts
/**
 * @file ISearchSpeciesUseCase.ts
 * @description Interfaz del caso de uso para buscar especies de plantas.
 * @module Plants
 * @layer Domain
 */

import type { PlantSpeciesResponseDTO } from '../../../dtos/plants/plant-species-response.dto.js';

export interface ISearchSpeciesUseCase {
  /**
   * @param query — Texto de búsqueda (nombre o nombre científico).
   * @returns Lista de especies públicas coincidentes como DTOs.
   */
  execute(query: string): Promise<PlantSpeciesResponseDTO[]>;
}
